internal/host/bruteforce/bruteforce.go

```diff
@@ -0,0 +1,200 @@
+package host
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/Method-Security/networkscan/generated/go/bruteforce"
+	modules "github.com/Method-Security/networkscan/internal/host/bruteforce/modules"
+)
+
+type BruteforceLibrary interface {
+	BruteForce(host string, port int, timeout int, options map[string]interface{}) (*bruteforce.AttemptInfo, []string)
+	AnalyzeResponse(attempt bruteforce.AttemptInfo) *bruteforce.ResultInfo
+	StandardPorts() []int
+}
+
+type BruteforceEngine struct {
+	Library BruteforceLibrary
+}
+
+func (be *BruteforceEngine) Run(ctx context.Context, subStep map[string]interface{}, target string, config *bruteforce.BruteForceRunConfig) ([]*bruteforce.AttemptInfo, []string) {
+	var attempts []*bruteforce.AttemptInfo
+	errors := []string{}
+
+	// If target is not a socket, use Module standard ports
+	var ports []int
+	host, port, err := net.SplitHostPort(target)
+	if err != nil {
+		host = target
+		ports = be.Library.StandardPorts()
+	} else {
+		intPort, err := strconv.Atoi(port)
+		if err != nil {
+			errors = append(errors, err.Error())
+			return nil, errors
+		}
+		ports = []int{intPort}
+	}
+
+	for _, port := range ports {
+		for i := 0; i < config.Retries; i++ {
+			attempt, errs := be.Library.BruteForce(host, port, config.Timeout, subStep)
+			if err != nil {
+				errors = append(errors, errs...)
+			}
+			// Check SuccessfulOnly config
+			if attempt.Result.Login || !config.SuccessfulOnly {
+				attempts = append(attempts, attempt)
+			}
+			// Break if credentials successful
```

Comment:
No, it breaks out of the ports loop not the creds loop but I guess if there are 2 ssh ports open on the standard port it could miss the other

Ack on flag to stop on first success